### PR TITLE
doc:  Fixup remove 'omitted...' doc for rpc getrawtransaction when verbose is 2

### PR DESCRIPTION
**Pull request by dougEfresh**
Remove optional rpc doc for getrawtransaction when verbose is 2

### CODE REVIEW

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -210,13 +210,13 @@ static RPCHelpMan getrawtransaction()
                         RPCResult::Type::OBJ, "", "",
                         {
                             {RPCResult::Type::ELISION, "", "Same output as verbosity = 1"},
-                            {RPCResult::Type::NUM, "fee", /*optional=*/true, "transaction fee in " + CURRENCY_UNIT + ", omitted if block undo data is not available"},
+                            {RPCResult::Type::NUM, "fee", "Transaction fee in " + CURRENCY_UNIT},
```
Comment (stickies-v) [inline src/rpc/rawtransaction.cpp:213]: The current wording and optionality seems to be correct to me, I [think](https://github.com/bitcoin/bitcoin/blob/79e007d1d60b8d600479d787667d91e05a492ab4/src/core_write.cpp#L254-L258) the fee is only included if undo data is present?

File: src/rpc/rawtransaction.cpp [hunk 2]
```diff
@@ -210,13 +210,13 @@ static RPCHelpMan getrawtransaction()
...
                             {RPCResult::Type::ARR, "vin", "",
                             {
-                                {RPCResult::Type::OBJ, "", "utxo being spent, omitted if block undo data is not available",
+                                {RPCResult::Type::OBJ, "", "utxo being spent",
                                 {
                                     {RPCResult::Type::ELISION, "", "Same output as verbosity = 1"},
-                                    {RPCResult::Type::OBJ, "prevout", /*optional=*/true, "Only if undo information is available)",
+                                    {RPCResult::Type::OBJ, "prevout", "The previous transaction",
```
Comment (stickies-v) [inline src/rpc/rawtransaction.cpp:219]: "prevout" refers to output, not a transaction. Also here, `/*optional=*/true` and `Only if undo information is available` [seem correct](https://github.com/bitcoin/bitcoin/blob/79e007d1d60b8d600479d787667d91e05a492ab4/src/core_write.cpp#L211-L228) to me, I'm open to adding a description of prevout being the previous output.

**Review by stickies-v — CHANGES_REQUESTED** [review]

**Comment by willcl-ark** [comment]
I agree with @stickies-v's comments here. We are not going to be returning those fields unless the undo data is available (for the previous coin), so I think the wording is correct as-is?

**Comment by stickies-v** [comment]
> so I think the wording is correct as-is?

Not sure if you're implying a NACK with this, but [this wording](https://github.com/bitcoin/bitcoin/pull/26968/files#diff-a58e7bb9d9a8a0287c0b7281d99da4e79b6f8c2a5780c24c6d76c14212c48640L216-R216) (i.e. the one initially highlighted by MarcoFalke) is incorrect imo and should indeed be fixed by this PR. `vin` is also returned if no undo data, it just won't contain a `prevout` field.

Concept ACK (so our beloved bot doesn't misinterpret me)

**Comment by willcl-ark** [comment]
> > so I think the wording is correct as-is?
> 
> Not sure if you're implying a NACK with this, but [this wording](https://github.com/bitcoin/bitcoin/pull/26968/files#diff-a58e7bb9d9a8a0287c0b7281d99da4e79b6f8c2a5780c24c6d76c14212c48640L216-R216) (i.e. the one initially highlighted by MarcoFalke) is incorrect imo and should indeed be fixed by this PR. `vin` is also returned if no undo data, it just won't contain a `prevout` field.

Right. Sorry I wasn't clear; I'm agreeing with you on both counts: 1 wording item to be changed (L216), 2 wording items correct as they are (L213 & L219).

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (dougEfresh) [reply src/rpc/rawtransaction.cpp:213]: The entire block is included only if undo data is present

Comment (stickies-v) [reply src/rpc/rawtransaction.cpp:213]: "fee" is not part of the "prevout" block, it is its own, top-level KV pair

**Comment by fanquake** [comment]
@dougEfresh are you going to followup here?

**Comment by dougEfresh** [comment]
> 

> up for grabs?

@MarcoFalke on it. Just been super busy last few weeks.

**Comment by dougEfresh** [comment]
@stickies-v This better?

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (stickies-v) [reply src/rpc/rawtransaction.cpp:216]: @dougEfresh this seems unaddressed?

File: src/rpc/rawtransaction.cpp (hunk 2 above)
Comment (stickies-v) [reply src/rpc/rawtransaction.cpp]: @dougEfresh this seems unaddressed?

**Comment by stickies-v** [comment]
I've provided links [[1](https://github.com/bitcoin/bitcoin/pull/26968#discussion_r1087741997), [2](https://github.com/bitcoin/bitcoin/pull/26968#discussion_r1087746520)] in my previous comments to reference the code from which I've built my understanding about which fields are optional and which ones aren't. Are we disagreeing about behaviour?  My understanding in short is: "vin" is always present for verbosity>=1, whereas "fee" and "prevout" are omitted if no block data.

So to me it seems like the latest force push (0404143) removes the only change that [actually had to be made](https://github.com/bitcoin/bitcoin/pull/26734#discussion_r1056512173) and what's left is incorrect.

I've committed the changes I think need to be made to https://github.com/stickies-v/bitcoin/commit/3280c03cc1dc412f2719de0ecf926eb1d70a0679 - feel free to use.

**Comment by dougEfresh** [comment]
@stickies-v I agree with your change, have pushed it to this PR. 
Thanks

**Comment by stickies-v** [comment]
I think ci is failing because you didn't fully apply my commit:

<details>
<summary>git diff 3280c03  3ecb9458c05ac869bc2d77ea983bc85baf9baa07 -- src/rpc/rawtransaction.cpp</summary>

```diff
diff --git a/src/rpc/rawtransaction.cpp b/src/rpc/rawtransaction.cpp
index 21d49fda9..986e7c0cc 100644
--- a/src/rpc/rawtransaction.cpp
+++ b/src/rpc/rawtransaction.cpp
@@ -219,7 +219,7 @@ static RPCHelpMan getrawtransaction()
                                 {RPCResult::Type::OBJ, "", "utxo being spent",
                                 {
                                     {RPCResult::Type::ELISION, "", "Same output as verbosity = 1"},
-                                    {RPCResult::Type::OBJ, "prevout", /*optional=*/true, "The previous output, omitted if block undo data is not available",
+                                    {RPCResult::Type::OBJ, /*optional=*/true, "The previous output, omitted if block undo data is not available",
                                     {
                                         {RPCResult::Type::BOOL, "generated", "Coinbase or not"},
                                         {RPCResult::Type::NUM, "height", "The height of the prevout"},
```diff
</summary>

**Review by stickies-v — APPROVED** [review]
ACK 3e947d7117c97a3cc34cfa7e1f5515fa0192fbe7

---

_Note: I'm glad that this is now in good shape, but for full transparency: to me this was a very frustrating review process and even though I hold no personal grudges over it, this will likely affect my enthusiasm towards reviewing your future work and for that reason I want to be transparent and share my feedback with you. I spent a lot of time writing out my thoughts and reasoning with links etc across multiple reviews, and yet every additional force-push felt like a "throw something at the wall and see if it sticks" approach. I have no problem whatsoever with clarifying misunderstanding or disagreement, but that was not the case - my comments were largely just ignored/unanswered. For such a trivial change, I spent way too much review time hand-holding._

**Pull request merged**